src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectSettingsRemoteAjaxController.java

```diff
@@ -0,0 +1,136 @@
+package ca.corefacility.bioinformatics.irida.ria.web.projects.settings;
+
+import java.security.Principal;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import ca.corefacility.bioinformatics.irida.model.RemoteAPI;
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.project.ProjectSyncFrequency;
+import ca.corefacility.bioinformatics.irida.model.remote.RemoteStatus;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.RemoteProjectSettings;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.RemoteProjectSettingsUpdateRequest;
+import ca.corefacility.bioinformatics.irida.service.ProjectService;
+import ca.corefacility.bioinformatics.irida.service.remote.ProjectRemoteService;
+import ca.corefacility.bioinformatics.irida.service.user.UserService;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Controller for managing settings for a remotely sync'd project
+ * asynchronously
+ */
+@RestController
+@RequestMapping("/ajax/remote-projects/{projectId}/settings")
+public class ProjectSettingsRemoteAjaxController {
+	private MessageSource messageSource;
+	private ProjectService projectService;
+	private ProjectRemoteService projectRemoteService;
+	private UserService userService;
+
+	@Autowired
+	public ProjectSettingsRemoteAjaxController(MessageSource messageSource, ProjectService projectService,
+			ProjectRemoteService projectRemoteService, UserService userService) {
+		this.messageSource = messageSource;
+		this.projectService = projectService;
+		this.projectRemoteService = projectRemoteService;
+		this.userService = userService;
+	}
+
+	/**
+	 * Update the project sync settings
+	 *
+	 * @param projectId                          the project id to update
+	 * @param remoteProjectSettingsUpdateRequest object which
+	 *                                           is used to update frequency, and sync user, as well as
+	 *                                           force sync for a project
+	 * @param principal                          The current logged in user
+	 * @param locale                             user's locale
+	 * @return result message if successful
+	 */
+	@PostMapping(value = "/sync")
+	public Map<String, String> updateProjectSyncSettings(@PathVariable Long projectId,
```

Comment:
You should return an object not a Map

Reply:
Updated in 3bd93720a5eb17ec1c86cc4c22cfa47f30d13312